code/messaging/src/test/java/com/adobe/marketing/mobile/messaging/ContentCardUIProviderTests.kt

```diff
@@ -0,0 +1,262 @@
+        whenever(contentCardSchemaData.content).thenReturn(contentCardMap)
+        whenever(contentCardSchemaData.meta).thenReturn(metaMap)
+        whenever(propositionItem.proposition).thenReturn(proposition)
+        whenever(proposition.uniqueId).thenReturn("testId")
+        contentCardSchemaData.parent = propositionItem
+        whenever(propositionItem.contentCardSchemaData).thenReturn(contentCardSchemaData)
+        whenever(proposition.items).thenReturn(listOf(propositionItem))
+    }
+
+    @Test
+    fun `getContentCardUI returns success with valid template`() = runTest {
+
+        mockMessaging.`when`<Unit> {
+            Messaging.getPropositionsForSurfaces(any(), any())
+        }.thenAnswer { invocation ->
+            val callback = invocation.arguments[1] as AdobeCallbackWithError<Map<Surface, List<Proposition>>>
+            callback.call(mapOf(surface to listOf(proposition)))
+        }
+
+        val flow = contentCardUIProvider.getContentCardUI()
+        val result = flow.first()
+
+        assertTrue(result.isSuccess)
+        assertNotNull(result.getOrNull())
+        assertTrue(result.getOrNull()?.isNotEmpty() == true)
+    }
+
+    @Test
+    fun `getContentCardUI handles null proposition map`() = runTest {
+        mockMessaging.`when`<Unit> {
+            Messaging.getPropositionsForSurfaces(any(), any())
+        }.thenAnswer { invocation ->
+            val callback = invocation.arguments[1] as AdobeCallbackWithError<Map<Surface, List<Proposition>>>
+            callback.call(null)
+        }
+
+        val flow = contentCardUIProvider.getContentCardUI()
+        val result = flow.first()
+        assertTrue(result.isFailure)
+    }
+
+    @Test
+    fun `getContentCardUI handles empty proposition list`() = runTest {
+        mockMessaging.`when`<Unit> {
+            Messaging.getPropositionsForSurfaces(any(), any())
+        }.thenAnswer { invocation ->
+            val callback = invocation.arguments[1] as AdobeCallbackWithError<Map<Surface, List<Proposition>>>
+            callback.call(mapOf(surface to emptyList()))
+        }
+
+        val flow = contentCardUIProvider.getContentCardUI()
+        val result = flow.first()
+        assertTrue(result.isSuccess)
+        assertTrue(result.getOrNull()?.isEmpty() == true)
+    }
+
+    @Test
+    fun `getContentCardUI handles invalid schema type`() = runTest {
+        whenever(propositionItem.schema).thenReturn(SchemaType.HTML_CONTENT)
+
+        mockMessaging.`when`<Unit> {
+            Messaging.getPropositionsForSurfaces(any(), any())
+        }.thenAnswer { invocation ->
+            val callback = invocation.arguments[1] as AdobeCallbackWithError<Map<Surface, List<Proposition>>>
+            callback.call(mapOf(surface to listOf(proposition)))
+        }
+
+        val flow = contentCardUIProvider.getContentCardUI()
+        val result = flow.first()
+        assertTrue(result.isSuccess)
```

Comment:
nit: Wondering if we isFailure should also be called here. Right now in `buildTemplate` we return null if the schema type is not as expected. Wondering if we should throw the error to be appended to list of error send in `result.isFailure`

Reply:
Same suggestion for other cases we return null in `ContentCardSchemaDataUtils` , totally optional though